File: 04_strings/ex02_inverter_string.py
"""
Exercicio 02
Tema: 04_strings
"""

def inverter(nome):
    letras = list(nome)
    inicio = 0
    fim = len(nome) - 1

    while inicio < fim:
        letras[inicio], letras[fim] = letras[fim], letras[inicio]
        inicio += 1
        fim -= 1
    
    return "".join(letras)

nome = "Nicolas"

inverter(nome)

File: 04_strings/ex02_inverter_string.c
/*
Exercicio 02
Tema: 04_strings
*/

#include <stdio.h>
#include <string.h>

void inverter_str(char * string){
    int inicio = 0;
    int fim = strlen(string) - 1;

    for(int i = 0; i < strlen(string)/2; i++){
        char temp = string[inicio];
        string[inicio] = string[fim];
        string[fim] = temp;

        inicio++;
        fim--;
    }
}

int main() {

    char string[20] = "Nicolas";

    inverter_str(string);

    printf("%s", string);

    return 0;
}
